resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/config.scala

```diff
@@ -0,0 +1,175 @@
+    ConfigBuilder("spark.kubernetes.namespace")
+      .doc("""
+          | The namespace that will be used for running the driver and
+          | executor pods. When using spark-submit in cluster mode,
+          | this can also be passed to spark-submit via the
+          | --kubernetes-namespace command line argument.
+        """.stripMargin)
+      .stringConf
+      .createWithDefault("default")
+
+  private[spark] val DRIVER_DOCKER_IMAGE =
+    ConfigBuilder("spark.kubernetes.driver.docker.image")
+      .doc("""
+          | Docker image to use for the driver. Specify this using the
+          | standard Docker tag format.
+        """.stripMargin)
+      .stringConf
+      .createWithDefault(s"spark-driver:$sparkVersion")
+
+  private[spark] val EXECUTOR_DOCKER_IMAGE =
+    ConfigBuilder("spark.kubernetes.executor.docker.image")
+      .doc("""
+          | Docker image to use for the executors. Specify this using
+          | the standard Docker tag format.
+        """.stripMargin)
+      .stringConf
+      .createWithDefault(s"spark-executor:$sparkVersion")
+
+  private[spark] val KUBERNETES_CA_CERT_FILE =
+    ConfigBuilder("spark.kubernetes.submit.caCertFile")
+      .doc("""
+          | CA cert file for connecting to Kubernetes over SSL. This
+          | file should be located on the submitting machine's disk.
+        """.stripMargin)
+      .stringConf
+      .createOptional
+
+  private[spark] val KUBERNETES_CLIENT_KEY_FILE =
+    ConfigBuilder("spark.kubernetes.submit.clientKeyFile")
+      .doc("""
+          | Client key file for authenticating against the Kubernetes
+          | API server. This file should be located on the submitting
+          | machine's disk.
+        """.stripMargin)
+      .stringConf
+      .createOptional
+
+  private[spark] val KUBERNETES_CLIENT_CERT_FILE =
+    ConfigBuilder("spark.kubernetes.submit.clientCertFile")
+      .doc("""
+          | Client cert file for authenticating against the
+          | Kubernetes API server. This file should be located on
+          | the submitting machine's disk.
+        """.stripMargin)
+      .stringConf
+      .createOptional
+
+  private[spark] val KUBERNETES_SERVICE_ACCOUNT_NAME =
+    ConfigBuilder("spark.kubernetes.submit.serviceAccountName")
+      .doc("""
+          | Service account that is used when running the driver pod.
+          | The driver pod uses this service account when requesting
+          | executor pods from the API server.
+        """.stripMargin)
+      .stringConf
+      .createWithDefault("default")
+
+  private[spark] val KUBERNETES_DRIVER_UPLOAD_JARS =
+    ConfigBuilder("spark.kubernetes.driver.uploads.jars")
+      .doc("""
+          | Comma-separated list of jars to sent to the driver and
+          | all executors when submitting the application in cluster
+          | mode.
+        """.stripMargin)
+      .stringConf
+      .createOptional
+
+  // Note that while we set a default for this in practice, it's
```

Comment:
maybe instead of "in practice", you mean "during submission" ?  I was expecting a default here in the config object after reading the comment.